src/Symfony/Component/Utf8/CodePoints.php

```diff
@@ -0,0 +1,234 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Utf8;
+
+use Symfony\Component\Utf8\Exception\InvalidArgumentException;
+
+final class CodePoints implements GenericStringInterface, \Countable
```

Comment:
I suggested this because I occasionally need to do checks where the built-in PHP features look confusing (getting a file extension, using `strpos()`, etc.):

```php
if (startsWith('https://', $url))
if (endsWith('.md', $file))
```

Symfony itself uses this a lot:

```php
// before
if (0 !== strpos($dsn, 'redis://'))
if (0 !== strpos($argument[0], '@') || 0 === strpos($argument[0], '@@'))

// after
if (startsWith('redis://', $dsn))
if (startsWith('@', $argument[0]) || !startsWith('@@', $argument[0]))
```

---

Probably out of the scope, but we could even accept arrays for both the prefix and the strings. The prefix logic would be `OR` (return true if starts/ends with either of the prefixes) and the strings logic would be `AND` (return true if all strings start/end with the prefix/prefixes):

```php
startsWith(string|array $prefix, string|array $strings)
endsWith(string|array $prefix, string|array $strings)
```

For example, this code in Symfony checks if the resource has either  the `yaml` or `yml` extensions:

```php
// before
in_array(pathinfo($resource, PATHINFO_EXTENSION), array('yaml', 'yml'), true));
```

It could be refactored as:

```php
// after
endsWith(['yaml', 'yml'], $resource);
```

Other examples:

```php
startsWith('memcached://', $lockStorages)
startsWith(['//', 'https://'], $assetUrls)
```